lib/data.js

```diff
@@ -24,47 +24,97 @@ api.common.copy = (
   return result;
 */
 
-api.common.clone = (
-  // Clone dataset (clone objects to new array)
-  ds
-) => cloneArray(ds);
+api.common.clone = (obj) => api.common.data.clone(obj);
+
+if (!api.common.data) api.common.data = {};
```

Comment:
`api.common.data = api.common.data || {};` but here we know that no `api.common.data` defined, so we can just use: `api.common.data = {};`